xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/url/resources/DocumentRestURLGenerator.java

```diff
@@ -44,6 +46,13 @@ public URL getURL(DocumentReference reference) throws XWikiRestException
         // The idea is to use the UriBuilder of jax-rs to generate URLs that match the resources paths.
         // So it is consistent.
         try {
+            Locale locale = reference.getLocale();
+            // Get the URL of the translated document if possible
```

Comment:
I would replace "if possible" by "if the passed reference defines a Locale". WDYT?